eclair-core/src/main/scala/fr/acinq/eclair/channel/Channel.scala

```diff
@@ -1678,16 +1678,15 @@ class Channel(val nodeParams: NodeParams, val wallet: EclairWallet, remoteNodeId
         cancelTimer(RevocationTimeout.toString)
       }
 
-      // if channel is private, we send the channel_update directly to remote
-      // they need it "to learn the other end's forwarding parameters" (BOLT 7)
-      (stateData, nextStateData) match {
-        case (d1: DATA_NORMAL, d2: DATA_NORMAL) if !d1.commitments.announceChannel && !d1.buried && d2.buried =>
-          // for a private channel, when the tx was just buried we need to send the channel_update to our peer (even if it didn't change)
+      (state, nextState, stateData, nextStateData) match {
+        // if channel is private, we send the channel_update directly to remote on each reconnect
+        // they need it "to learn the other end's forwarding parameters" (BOLT 7)
+        case (NORMAL, NORMAL, d1: DATA_NORMAL, d2: DATA_NORMAL) if !d1.commitments.announceChannel && !d1.buried && d2.buried =>
           forwarder ! d2.channelUpdate
-        case (d1: DATA_NORMAL, d2: DATA_NORMAL) if !d1.commitments.announceChannel && d1.channelUpdate != d2.channelUpdate && d2.buried =>
-          // otherwise, we only send it when it is different, and tx is already buried
+        case (OFFLINE, NORMAL, d1: DATA_NORMAL, d2: DATA_NORMAL) if !d1.commitments.announceChannel && d2.buried =>
```

Comment:
I believe you still need the other case: when you're in state normal and update your fee policy, you do want to re-broadcast the new `channel_update`.

I'm also not sure the case you added will ever happen, because you're using `d1: DATA_NORMAL` from state `OFFLINE`...that requires more tests but doesn't look correct.

Reply:
Unless I miss something `DATA_NORMAL` is the one which present in `OFFLINE` state and `OFFLINE -> NORMAL` transition?

Reply:
Woops you're totally right, I'll have another look at that.